client/src/components/AppBar.jsx

```diff
@@ -0,0 +1,111 @@
+import React, { useState } from "react";
+import "./AppBar.css";
+
+// Validation utility function
+const isValidInput = (title, url) => {
+  const isValidTitle = typeof title === "string" && title.trim() !== "";
+  const isValidUrl =
+    typeof url === "string" &&
+    /^(https?:\/\/)?(www\.)?(youtube\.com\/watch\?v=|youtube\.be\/)[\w-]{11}$/.test(
+      url
+    );
+  return isValidTitle && isValidUrl;
+};
+function AppBar({ onAddCard, cards }) {
```

Comment:
Try and put an empty line between functions, it makes it a bit easier to read!
But great job seperating out this log from your component 😄